aspnetcore/razor-components/forms-validation.md

```diff
@@ -0,0 +1,181 @@
+    <InputText id="name" bind-Value="@exampleModel.Name" />
+
+    <button type="submit">Submit</button>
+</EditForm>
+
+@functions {
+    private ExampleModel exampleModel = new ExampleModel();
+
+    void HandleValidSubmit()
+    {
+        Console.WriteLine("OnValidSubmit");
+    }
+}
+```
+
+* The form validates user input in the `name` field using the validation defined in the `ExampleModel` type. The model is created in the component's `@functions` block and held in a private field (`exampleModel`). The field is assigned to the `Model` attribute of the `<EditForm>`.
+* The Data Annotations Validator component (`<DataAnnotationsValidator>`) attaches validation support using data annotations.
+* The Validation Summary component (`<ValidationSummary>`) summarizes validation messages.
+* `HandleValidSubmit` is triggered when the form successfully submits (passes validation).
+
+A set of built-in input components are available to receive and validate user input. Inputs are validated when they're changed and when a form is submitted. Available input components are shown in the following table.
+
+| Input component   | Rendered as&hellip;       |
+| ----------------- | ------------------------- |
+| `<InputText>`     | `<input>`                 |
+| `<InputTextArea>` | `<textarea>`              |
+| `<InputSelect>`   | `<select>`                |
+| `<InputNumber>`   | `<input type="number">`   |
+| `<InputCheckbox>` | `<input type="checkbox">` |
+| `<InputDate>`     | `<input type="date">`     |
+
+Input components provide default behavior for validating on edit and changing their CSS class to reflect the field state. Some components include useful parsing logic. For example, `<InputDate>` and `<InputNumber>` handle unparseable values gracefully by registering them as validation errors. Types that can accept null values also support nullability of the target field (for example, `int?`).
+
+The following `Starship` type defines validation logic using a larger set of properties and [data annotations](xref:mvc/models/validation) than the earlier `ExampleModel`:
+
+```csharp
+using System;
+using System.ComponentModel.DataAnnotations;
+
+public class Starship
+{
+    [Required]
+    [StringLength(16, 
+        ErrorMessage = "Identifier too long (16 character limit).")]
+    public string Identifier { get; set; }
+
+    // Optional (no data annotations)
+    public string Description { get; set; }
+
+    [Required]
+    public string Classification { get; set; }
+
+    [Range(1, 100000, ErrorMessage = "Accommodation invalid (1-100000).")]
+    public int MaximumAccommodation { get; set; }
+
+    [Required]
+    [Range(typeof(bool), "true", "true", 
+        ErrorMessage = "Form disallowed for unapproved ships.")]
+    public bool IsValidatedDesign { get; set; }
+
+    [Required]
+    public DateTime ProductionDate { get; set; }
+}
+```
+
+The following Starship Database form validates user input using the validation defined in the `Starship` model:
```

Comment:
The following form validates user input using the validation defined in the `Starship` model:

Reply:
The `Starship` model :smile:. HA! Lov'in it! :heart: